fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -316,7 +316,7 @@ public Void visitArrayExpr(ArrayExpr node, Scope scope) {
                     throw new SemanticException(e.getMessage());
                 }
             } else {
-                node.setType(new ArrayType(Type.NULL));
```

Comment:
can you add some commet here which corresponds to the comment in ScalarOperatorToExpr